src/main/java/org/broadinstitute/hellbender/tools/spark/sv/SVFastqUtils.java

```diff
@@ -4,49 +4,245 @@
+        private final int start;
+
+        private final boolean forwardStrand;
+
+        private final Cigar cigar;
+
+        public Mapping(final GATKRead read) {
+            Utils.nonNull(read);
+            if (read.isUnmapped()) {
+                contig = null;
+                start = -1;
+                forwardStrand = true;
+                cigar = null;
+            } else {
+                contig = read.getContig();
+                start = read.getStart();
+                forwardStrand = !read.isReverseStrand();
+                cigar = read.getCigar();
+            }
+        }
+
+        public Mapping(final String mappingDescription) {
+            Utils.nonNull(mappingDescription);
+            if (mappingDescription.equals(UNMAPPED_DESCRIPTION_STR)) {
+                contig = null;
+                start = -1;
+                forwardStrand = true;
+                cigar = null;
+            } else {
+                final Matcher matcher = MAPPING_DESCRIPTION_PATTERN.matcher(mappingDescription);
+                if (!matcher.find()) {
+                    throw new IllegalArgumentException("invalid mapping description: '" + mappingDescription + "'");
+                } else {
+                    contig = matcher.group(1);
+                    start = Integer.parseInt(matcher.group(2));
+                    cigar = TextCigarCodec.decode(matcher.group(4));
+                    forwardStrand = Strand.decode(matcher.group(3)) == Strand.POSITIVE;
+                }
+            }
+        }
+
+        public boolean isMapped() {
+            return contig != null;
+        }
+
+        @Override
+        public String getContig() {
+            return contig;
+        }
+
+        @Override
+        public int getStart() {
+            return start;
+        }
+
+        @Override
+        public int getEnd() {
+            return start;
+        }
+
+        public Cigar getCigar() {
+            return cigar;
+        }
+
+        public boolean isForwardStrand() {
+            return forwardStrand;
+        }
+
+        public String toString() {
+            if (isMapped()) {
+                return MAPPING_FIELD_EQUAL_TO + String.join(HEADER_FIELD_ARRAY_SEPARATOR_STR, getContig() + SimpleInterval.CONTIG_SEPARATOR + getStart(), (forwardStrand ? Strand.POSITIVE : Strand.NEGATIVE).encodeAsString(), cigar.toString());
+            } else {
+                return UNMAPPED_DESCRIPTION_STR;
+            }
+        }
+    }
+
     @DefaultSerializer(FastqRead.Serializer.class)
     public static final class FastqRead implements FermiLiteAssembler.BasesAndQuals {
-        private final String name;
+        private final String header;
         private final byte[] bases;
         private final byte[] quals;
 
-        public FastqRead( final String name, final byte[] bases, final byte[] quals ) {
-            this.name = name;
+        public FastqRead( final GATKRead read ) {
+            this(Utils.nonNull(read).getName(),
```

Comment:
I'd rather see simple inline code here to initialize the three fields correctly, rather than calling another constructor with a bunch of ternary operators.  You can then also eliminate reverseIf, composeHeaderLine and the next two constructors that are otherwise unused.  I think it would be much clearer that way.  But this is a matter of personal taste, I suppose, so if you like it this way better I won't object.